### ISSUE TITLE
Smart search returns incorrect number of items

### ISSUE THREAD

**Issue by thanhnv37** [joomla/joomla-cms#25816]
### Steps to reproduce the issue
In the Finder indexing plugin, I set multiple values for one taxonomy as following:

```
$item->addTaxonomy('Variants', 'catalan');
$item->addTaxonomy('Variants', 'occitan');
$item->addTaxonomy('Variants', 'test');
```

### Expected result
On the search page, the item must be returned on search results if I filter by Variants field with any its variants.

### Actual result
Some items are missing on search results.

For more details:
If I filter by Variant "occitan",
The results total (in pagination) display correct number of items (E.g: 1-14 of 14)
But the search results displays only 6 items.

### System information (as much as possible)
After some investigating, I can fix this issue with solution below:
1. Open the file: 
**components/com_finder/models/search.php**

2. Go to the code line 226:
`->select('l.link_id')`

3. Replace it with below one:
`->select('DISTINCT l.link_id')`

4. Save changes and the search filter works worked correctly.
### Additional comments
I hope that you can apply this fix on next release of Joomla.
Thanks

**Comment by SharkyKZ** [comment]
Test PR #23735 please.

**Comment by ghost** [comment]
Closed as having Pull Request #23735
